runtime/src/lib.rs

```diff
@@ -256,7 +291,18 @@ fn handle_command(drivers: &mut Drivers) -> CaliptraResult<MboxStatusE> {
         #[cfg(feature = "test_only_commands")]
         CommandId::TEST_ONLY_HMAC384_VERIFY => HmacVerifyCmd::execute(drivers, cmd_bytes),
         CommandId::VERSION => FipsVersionCmd::execute(drivers),
-        CommandId::SELF_TEST => FipsSelfTestCmd::execute(drivers),
+        #[cfg(feature = "fips_self_test")]
+        CommandId::SELF_TEST => match drivers.self_test_status {
```

Comment:
I don't think you should overload the SELF_TEST command for the "start" and "get result" phases. The response in both cases is an empty success result, which isn't very informative. See my comments in smoke_test.rs.

Reply:
done